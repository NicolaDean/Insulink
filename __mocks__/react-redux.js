// This mock will make sure that we are able to access mapStateToProps, mapDispatchToProps and reactComponent in the test file.

// To use this, just do `jest.mock('react-redux');` in your page.test.js file.
const mockDispatch = jest.fn((action) => action);

module.exports = {
  useDispatch: () => mockDispatch,
  connect: (mapStateToProps, mapDispatchToProps) => (reactComponent) => ({
    mapStateToProps,
    mapDispatchToProps: (dispatch = mockDispatch, ownProps) => mapDispatchToProps(dispatch, ownProps),
    reactComponent,
    mockDispatch
  }),
  Provider: ({children}) => children
};