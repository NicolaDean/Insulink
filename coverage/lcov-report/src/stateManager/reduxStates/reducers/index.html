
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/stateManager/reduxStates/reducers</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../../index.html">All files</a> src/stateManager/reduxStates/reducers</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">84.21% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>144/171</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">54.34% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>25/46</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">73.91% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>17/23</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">84.61% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>143/169</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	</tr>

<tr>
	<td class="file high" data-value="macroTracker.js"><a href="macroTracker.js.html">macroTracker.js</a></td>
	<td data-value="88.61" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 88%"></div><div class="cover-empty" style="width: 12%"></div></div>
	</td>
	<td data-value="88.61" class="pct high">88.61%</td>
	<td data-value="123" class="abs high">109/123</td>
	<td data-value="52.94" class="pct medium">52.94%</td>
	<td data-value="34" class="abs medium">18/34</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="15" class="abs high">12/15</td>
	<td data-value="89.34" class="pct high">89.34%</td>
	<td data-value="122" class="abs high">109/122</td>
	</tr>

<tr>
	<td class="file medium" data-value="userReducer.js"><a href="userReducer.js.html">userReducer.js</a></td>
	<td data-value="72.34" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 72%"></div><div class="cover-empty" style="width: 28%"></div></div>
	</td>
	<td data-value="72.34" class="pct medium">72.34%</td>
	<td data-value="47" class="abs medium">34/47</td>
	<td data-value="58.33" class="pct medium">58.33%</td>
	<td data-value="12" class="abs medium">7/12</td>
	<td data-value="62.5" class="pct medium">62.5%</td>
	<td data-value="8" class="abs medium">5/8</td>
	<td data-value="71.73" class="pct medium">71.73%</td>
	<td data-value="46" class="abs medium">33/46</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Thu May 26 2022 11:26:03 GMT+0200 (Central European Summer Time)
            </div>
        <script src="../../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../../sorter.js"></script>
        <script src="../../../../block-navigation.js"></script>
    </body>
</html>
    