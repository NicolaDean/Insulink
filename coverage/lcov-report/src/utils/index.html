
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/utils</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> src/utils</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">51.97% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>184/354</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">20.16% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>24/119</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">50% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>37/74</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">52.02% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>180/346</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line medium'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="apiHelper.js"><a href="apiHelper.js.html">apiHelper.js</a></td>
	<td data-value="92.5" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 92%"></div><div class="cover-empty" style="width: 8%"></div></div>
	</td>
	<td data-value="92.5" class="pct high">92.5%</td>
	<td data-value="40" class="abs high">37/40</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="88.88" class="pct high">88.88%</td>
	<td data-value="9" class="abs high">8/9</td>
	<td data-value="92.5" class="pct high">92.5%</td>
	<td data-value="40" class="abs high">37/40</td>
	</tr>

<tr>
	<td class="file high" data-value="apiQuery.js"><a href="apiQuery.js.html">apiQuery.js</a></td>
	<td data-value="86.48" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 86%"></div><div class="cover-empty" style="width: 14%"></div></div>
	</td>
	<td data-value="86.48" class="pct high">86.48%</td>
	<td data-value="37" class="abs high">32/37</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="6" class="abs high">5/6</td>
	<td data-value="90" class="pct high">90%</td>
	<td data-value="10" class="abs high">9/10</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="35" class="abs high">30/35</td>
	</tr>

<tr>
	<td class="file low" data-value="auth.js"><a href="auth.js.html">auth.js</a></td>
	<td data-value="5.55" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 5%"></div><div class="cover-empty" style="width: 95%"></div></div>
	</td>
	<td data-value="5.55" class="pct low">5.55%</td>
	<td data-value="36" class="abs low">2/36</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="14" class="abs low">0/14</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="8" class="abs low">0/8</td>
	<td data-value="5.55" class="pct low">5.55%</td>
	<td data-value="36" class="abs low">2/36</td>
	</tr>

<tr>
	<td class="file low" data-value="chartDataFormatter.js"><a href="chartDataFormatter.js.html">chartDataFormatter.js</a></td>
	<td data-value="5.26" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 5%"></div><div class="cover-empty" style="width: 95%"></div></div>
	</td>
	<td data-value="5.26" class="pct low">5.26%</td>
	<td data-value="19" class="abs low">1/19</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="5.26" class="pct low">5.26%</td>
	<td data-value="19" class="abs low">1/19</td>
	</tr>

<tr>
	<td class="file low" data-value="firebaseQuery.js"><a href="firebaseQuery.js.html">firebaseQuery.js</a></td>
	<td data-value="46.34" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 46%"></div><div class="cover-empty" style="width: 54%"></div></div>
	</td>
	<td data-value="46.34" class="pct low">46.34%</td>
	<td data-value="82" class="abs low">38/82</td>
	<td data-value="3.84" class="pct low">3.84%</td>
	<td data-value="26" class="abs low">1/26</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="15" class="abs medium">9/15</td>
	<td data-value="48.05" class="pct low">48.05%</td>
	<td data-value="77" class="abs low">37/77</td>
	</tr>

<tr>
	<td class="file low" data-value="inputChecker.js"><a href="inputChecker.js.html">inputChecker.js</a></td>
	<td data-value="20.37" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 20%"></div><div class="cover-empty" style="width: 80%"></div></div>
	</td>
	<td data-value="20.37" class="pct low">20.37%</td>
	<td data-value="54" class="abs low">11/54</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="36" class="abs low">0/36</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	<td data-value="20.37" class="pct low">20.37%</td>
	<td data-value="54" class="abs low">11/54</td>
	</tr>

<tr>
	<td class="file medium" data-value="insulineCalculator.js"><a href="insulineCalculator.js.html">insulineCalculator.js</a></td>
	<td data-value="57.14" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 57%"></div><div class="cover-empty" style="width: 43%"></div></div>
	</td>
	<td data-value="57.14" class="pct medium">57.14%</td>
	<td data-value="21" class="abs medium">12/21</td>
	<td data-value="66.66" class="pct medium">66.66%</td>
	<td data-value="12" class="abs medium">8/12</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="5" class="abs low">2/5</td>
	<td data-value="57.14" class="pct medium">57.14%</td>
	<td data-value="21" class="abs medium">12/21</td>
	</tr>

<tr>
	<td class="file medium" data-value="localStoreManager.js"><a href="localStoreManager.js.html">localStoreManager.js</a></td>
	<td data-value="77.77" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 77%"></div><div class="cover-empty" style="width: 23%"></div></div>
	</td>
	<td data-value="77.77" class="pct medium">77.77%</td>
	<td data-value="63" class="abs medium">49/63</td>
	<td data-value="57.14" class="pct medium">57.14%</td>
	<td data-value="14" class="abs medium">8/14</td>
	<td data-value="56.25" class="pct medium">56.25%</td>
	<td data-value="16" class="abs medium">9/16</td>
	<td data-value="77.41" class="pct medium">77.41%</td>
	<td data-value="62" class="abs medium">48/62</td>
	</tr>

<tr>
	<td class="file high" data-value="testingJsons.js"><a href="testingJsons.js.html">testingJsons.js</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Thu May 26 2022 11:26:03 GMT+0200 (Central European Summer Time)
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    