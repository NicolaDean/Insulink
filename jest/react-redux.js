
const mockDispatch = jest.fn((action) => console.log(action));
//const removeFood= jest.spyOn(actions,'removeFood');

export const reduxMock = {
    useDispatch: () => mockDispatch,
    connect: (mapStateToProps, mapDispatchToProps) => (reactComponent) => ({
      mapStateToProps,
      mapDispatchToProps: (dispatch = mockDispatch, ownProps) => mapDispatchToProps(dispatch, ownProps),
      reactComponent,
      mockDispatch
    }),
    Provider: ({children}) => children,
    dispatch : mockDispatch,
    getMock: () => mockDispatch
  }

  export const baaa = {
    ciao:"aa",
    aa:"ciao"
  }