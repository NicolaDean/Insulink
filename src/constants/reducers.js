export const foodMethods = {
    addFood:'ADD_FOOD',
    removeFood:'REMOVE_FOOD',
    selectMeal:'SELECT_MEAL',
}

export const userMethods = {
    registerUser: 'REGISTER_USER',
    deleteUser: 'DELETE_USER',
    logout: 'LOGOUT_USER',
    login: 'LOGIN_USER',
    updateUser: 'UPDATE_USER',
    addGlicemy: 'ADD_GLICEMY',
}