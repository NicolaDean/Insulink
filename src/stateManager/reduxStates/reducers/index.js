import macroReducer from "./macroTracker";
import { combineReducers } from "redux";

const appReducers = combineReducers({
    macroTracker : macroReducer,
});

export default appReducers;