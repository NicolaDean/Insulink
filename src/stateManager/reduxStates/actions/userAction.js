import { firebase } from "@react-native-firebase/firestore";
import { userMethods } from "../../../constants/reducers"

import * as database from "../../../utils/firebaseQuery"
import * as localStorage from '../../../utils/localStoreManager'
import * as authSys from '../../../utils/auth'

/**
 * check all user inputs and try to register
 * if all ok send data to firebase and register
 * else return an error code or an error message to show user
 * @param {*} user user data to register
 * @returns error code
 */
export const register = (user,password) => async dispatch =>{
    //ASYNC ACTION:
    //TODO CHECK USER INPUTS ARE OK (psw length, email not exist etc...)

    console.log("Reg Start:");
    const id = (await authSys.register(user.email,password)).uid;

    console.log("Reg END:" + id);
    await database.registerUser(id,user);
    
    console.log("REGISTER OK");

    await localStorage.setDataAvailability(true);
    //SAVE DATA TO LOCAL STORAGE

    //REDUX DISPATCH
    dispatch({
        type: userMethods.registerUser,
        payload: {user:user}
    });
} 

/**
 * edit user data
 * @param {*} userData 
 * @returns 
 */
export const editUserData = (userData) => async dispatch =>{


    //Save updated user data to local storage
    //localStorage.saveUserData(userData);
    //Firebase Update
    //database.udpateUser(userData);

    console.log("Edit Completed");
    //REDUX DISPATCH
    dispatch({
        type: userMethods.updateUser,
        payload: {user:userData}
    });
}


export const del = (user) =>{

    //REDUX DISPATCH
    dispatch({
        type: userMethods.deleteUser,
        payload: {user:user}
    });
} 

/**
 * retrive user data from firebase, check correctness of login
 * if login is ok save data to local storage and into redux state
 * @param {*} email id of user
 * @param {*} psw password for login
 * @returns true or false (false if login is not ok)
 */
export const login = (email,psw) => async dispatch =>{

    //ASYNC ACTION (eg check values on DB)

    //CHECK LOGIN DATA
    const user = await authSys.login(email,psw);

    //Get user data
    const usrData = (await database.getUserData(user.uid));
    
    const glicemy = (await database.getUserGlicemy(user.uid));

    usrData.glicemy = glicemy;

    //CHECK FOR EMPTY DATA:
    usrData.age = 20; //TODO CALCULATE AGE FROM BIRTHDAY
    
    console.log("Login ok" + JSON.stringify(usrData));

    await localStorage.setDataAvailability(true);
    await localStorage.saveUserData(usrData);

    //if(login not good) return false;

    //REDUX DISPATCH
    dispatch({
        type: userMethods.login,
        payload: {
            usrData: usrData
        }
    });

    return true;
} 


/**
 * remove all user data from local storage and execute logout
 * @param {*} email id of user
 * @returns nothing
 */
export const logout = (email) => async dispatch =>{
    //ASYNC ACTION (eg check values on DB)

    await localStorage.resetUser();

    //REDUX DISPATCH
    dispatch({
        type: userMethods.logout
    });
}

/**
 * 
 * @param {*} email id of user
 * @param {*} glicemyValue an integer representing the value of glicemy in this specific moment
 * @returns nothing
 */
export const addGlicemy = (userId,glicemyValue) => async dispatch =>{

    const date = new Date();
    //GET TIMESTAMP
    const glicemy = {value: glicemyValue,time:date};

    console.log("G: " + JSON.stringify(glicemy));
    //UPDATE FIREBASE IF POSSIBLE; IF NOT ADD A PENDING UPDATE (TODO)
    database.addGlicemyValue(userId,glicemy);

    //REDUX DISPATCH
    dispatch({
        type:userMethods.addGlicemy,
        payload:{glicemy:glicemy}
    })
}