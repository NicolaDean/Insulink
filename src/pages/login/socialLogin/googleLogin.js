
import React,{useState} from 'react'
import { Button,View  } from 'react-native';

import { GoogleSignin,GoogleSigninButton } from '@react-native-google-signin/google-signin';
import auth from '@react-native-firebase/auth';
import { useDispatch } from 'react-redux';
import { googleLogin } from '../../../stateManager/reduxStates/actions/userAction';
import { RegistrationErrorPopup } from '../registration/errorsPopup';

GoogleSignin.configure({
  webClientId: '503937751847-f3p5g4fv6pa6l41v65avpobugi0f2bjd.apps.googleusercontent.com',
});

export async function onGoogleButtonPress() {

    try{
        GoogleSignin.signOut()
        if (GoogleSignin.isSignedIn()) {
            
        }

        // Get the users ID token
        const { idToken } = await GoogleSignin.signIn();
        
        // Create a Google credential with the token
        const googleCredential = auth.GoogleAuthProvider.credential(idToken);

        // Sign-in the user with the credential
        const x = await auth().signInWithCredential(googleCredential);
        //console.log(JSON.stringify(x));
        return x.user;
    } catch(e){
        console.log(console.log(e));
        if(e.code == 'ASYNC_OP_IN_PROGRESS'){
            console.log("ALREADY SINGED IN")
        }
        //TODO PUT THE OTHERS CODES
    };

    return null;
    
  }

export const GoogleButton = ({navigation,onSuccess}) =>{

    const dispatch = useDispatch();

    const [errors,setErrors] = useState([]);
    const [errorModalVisible, setErrorModalVisible] = useState(false);

    const errorFunc = (error) => {
        setErrors(error);

        if(JSON.stringify(error) != "[]") {
            setErrorModalVisible(true);
        }

        console.log(JSON.stringify(error))
    };
    const checkUserExistance = async() =>{
        //TODO check if exist the user in firebase side (otherwise redirect to Registration skipping page 1)
    
        const userData = await onGoogleButtonPress();
        
        if(userData!=null){
            console.log("Google Login UID: " + userData.uid);
            dispatch(googleLogin(userData.uid,errorFunc,navigation));
        }
        
    }

    return (
        <View>
            <RegistrationErrorPopup visibilityFlag={ [errorModalVisible, setErrorModalVisible]} errors={errors}/>
           
           <GoogleSigninButton
               style={{ width: 192, height: 48,alignSelf:'center' }}
               size={GoogleSigninButton.Size.Wide}
               color={GoogleSigninButton.Color.Dark}
               onPress={checkUserExistance}
           />
        </View>
       
      );
}