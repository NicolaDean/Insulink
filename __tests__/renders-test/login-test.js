import React from "react";
import {create,act} from 'react-test-renderer';
import { userMethods } from "../../src/constants/reducers";
import Home from "../../src/pages/home/home";
import { login } from "../../src/stateManager/reduxStates/actions/userAction";
import { mockedStore } from "../../testHelper/reduxMock";
import CustomFirestoreMock from "../../__mocks__/@react-native-firebase/firebase";
import { FirebaseQuery } from "../../src/utils/firebaseQuery";
import { Provider } from "react-redux";
import Login from "../../src/pages/login/login";
import * as Google from "../../src/pages/login/socialLogin/googleLogin";
import { configure } from "@testing-library/react";
import { NativeModules } from 'react-native';
import { GoogleSignin } from "@react-native-google-signin/google-signin";
//FIREBASE MOCK
const mock_firebase = new CustomFirestoreMock();
FirebaseQuery.users = mock_firebase;

  const firebasee = () =>{
      return {
          collection:jest.fn((a)=>{console.log("COLLECTION->"+ a);return mock_firebase}),
          doc:jest.fn((a)=>{console.log("DOC->"+ a);return mock_firebase})
      };
  }

  
  
  const userData = {name:"test",height:77,glicemy: []}
  const userId = "ABCDE123456";
  const action ={
      type:userMethods.login,
      payload:{
          usrData:userData,
          userId:userId
      }
  }

jest.mock('@react-native-firebase/firestore',()=> firebasee);
jest.mock('invariant/invariant');

//MOCK NAVIGATION
const mock_navigation = {
    navigate:jest.fn((a,b)=>{console.log("NAV to " + a)})
}

//MOCK GOOGLE SINGIN

const googleMock = jest.spyOn(Google,'onGoogleButtonPress');
googleMock.mockReturnValue = {
    uid:"ABCDE12345"
}

//TREE DECLARATION
var tree;

/*jest.mock("./AwesomeComponent", () => ({
    AwesomeComponent: () => {
      const MockName = "named-awesome-component-mock";
      return <MockName />;
    },
  }));*/


const mock_z = new yyyy();

GoogleSignin.configure = jest.fn(()=>{});
NativeModules.RNGoogleSignin = {
    BUTTON_SIZE_ICON: 0,
    BUTTON_SIZE_STANDARD: 0,
    BUTTON_SIZE_WIDE: 0,
    BUTTON_COLOR_AUTO: 0,
    BUTTON_COLOR_LIGHT: 0,
    BUTTON_COLOR_DARK: 0,
    SIGN_IN_CANCELLED: '0',
    IN_PROGRESS: '1',
    PLAY_SERVICES_NOT_AVAILABLE: '2',
    SIGN_IN_REQUIRED: '3',
    configure: jest.fn(),
    currentUserAsync: jest.fn(),
  };

  Google.GoogleSignin.hasPlayServices = () => Promise.resolve(true);
  Google.GoogleSignin.configure = () => Promise.resolve();
  Google.GoogleSignin.currentUserAsync = () => {
    return Promise.resolve({
      name: 'name',
      email: 'test@example.com',
      // .... other user data
    });
  };
jest.mock('@react-native-google-signin/google-signin', () => {
    const mockGoogleSignin = jest.requireActual('@react-native-google-signin/google-signin');
  
   
  
    // ... and other functions you want to mock
  
    return mockGoogleSignin;
  });


 // jest.mock('@react-native-firebase/auth');
  //jest.mock('@react-native-google-signin/google-signin');
  //jest.mock('../../src/pages/login/socialLogin/googleLogin');
  //jest.mock('@react-native-google-signin/google-signin', () => {});
 // jest.mock('node_modules/@react-native-google-signin/google-signin/');

describe("Test the Login page:",()=>{
    
    beforeAll(async ()=>{
        await act(async() => {
            tree = create(<Provider store={mockedStore}>
                            <Login  navigation={mock_navigation}/>
                          </Provider>)
          });

         
    });

    

    test("Error Display",()=>{

        expect(a).toHaveTextContent()
    })
})