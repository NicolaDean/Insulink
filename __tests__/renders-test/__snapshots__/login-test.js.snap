// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Login page: Loading of Page and Login Redirect 1`] = `
Object {
  "errorReducer": Object {
    "errorFunc": [Function],
    "reset": [Function],
  },
  "macroTracker": Object {
    "activities": Object {
      "breakfast": Object {
        "sports": Array [],
        "totCal": 0,
      },
      "dinner": Object {
        "sports": Array [],
        "totCal": 0,
      },
      "lunch": Object {
        "sports": Array [],
        "totCal": 0,
      },
      "snack": Object {
        "sports": Array [],
        "totCal": 0,
      },
    },
    "activity_id": 1,
    "currentDate": "[DATE]",
    "currentMeal": "breakfast",
    "history": Object {
      "activities": Object {
        "breakfast": Object {
          "sports": Array [],
          "totCal": 0,
        },
        "dinner": Object {
          "sports": Array [],
          "totCal": 0,
        },
        "lunch": Object {
          "sports": Array [],
          "totCal": 0,
        },
        "snack": Object {
          "sports": Array [],
          "totCal": 0,
        },
      },
      "empty": true,
      "meals": Object {
        "breakfast": Object {
          "foods": Array [],
          "macro": Object {
            "cal": 0,
            "carb": 0,
            "fat": 0,
            "prot": 0,
          },
        },
        "dinner": Object {
          "foods": Array [],
          "macro": Object {
            "cal": 0,
            "carb": 0,
            "fat": 0,
            "prot": 0,
          },
        },
        "lunch": Object {
          "foods": Array [],
          "macro": Object {
            "cal": 0,
            "carb": 0,
            "fat": 0,
            "prot": 0,
          },
        },
        "snack": Object {
          "foods": Array [],
          "macro": Object {
            "cal": 0,
            "carb": 0,
            "fat": 0,
            "prot": 0,
          },
        },
      },
      "totMacro": Object {
        "cal": 0,
        "carb": 0,
        "fat": 0,
        "prot": 0,
      },
    },
    "id": 1,
    "meals": Object {
      "breakfast": Object {
        "foods": Array [],
        "macro": Object {
          "cal": 0,
          "carb": 0,
          "fat": 0,
          "prot": 0,
        },
      },
      "dinner": Object {
        "foods": Array [],
        "macro": Object {
          "cal": 0,
          "carb": 0,
          "fat": 0,
          "prot": 0,
        },
      },
      "lunch": Object {
        "foods": Array [],
        "macro": Object {
          "cal": 0,
          "carb": 0,
          "fat": 0,
          "prot": 0,
        },
      },
      "snack": Object {
        "foods": Array [],
        "macro": Object {
          "cal": 0,
          "carb": 0,
          "fat": 0,
          "prot": 0,
        },
      },
    },
    "totCalBurned": 0,
    "totMacro": Object {
      "cal": 0,
      "carb": 0,
      "fat": 0,
      "prot": 0,
    },
  },
  "userReducer": Object {
    "mustCompleteReg": false,
    "status": "LOGGED",
    "userData": Object {
      "birthday": Object {
        "nanoseconds": 0,
        "seconds": 0,
      },
      "choratio": 0,
      "email": "test@a.com",
      "glicemy": Array [],
      "height": 180,
      "isf": 0,
      "maxCarb": 200,
      "maxFat": 100,
      "maxProt": 40,
      "name": "dummy",
      "password": "22222",
      "surname": "aaa",
      "weight": 80,
    },
    "userId": "ABCD12345",
  },
}
`;

exports[`Test the Login page: Login Button Click 1`] = `
Array [
  "Test",
  "Test",
  [Function],
]
`;
